### PR TITLE
Remove mount NTFS error message

### PR DESCRIPTION
**Pull request by KsenijaS**
Provide an option to suppress error logging from _mount_cb_ as some errors can be expected error and handled appropriately by DataSources. For example: failure to mount NTFS volumes on VMs that do not have NTFS drivers.

### CODE REVIEW

**Comment by cjp256** [comment]
When opening a PR, I would suggest squashing the branch to start the PR with the minimum number of commits (usually one) with a descriptive commit message(s).  It will reduce the work for the maintainer to finalize the commit message when merging.

**Comment by TheRealFalcon** [comment]
@cjp256 , do these changes look reasonable to you as they're Azure-specific? Is there any time where we would want the log but won't see it?

**Comment by anhvoms** [comment]
> Remove error message from cloud-init log when trying to mount ntfs on Redhat/CentOS. This is an expected behavior.

Suggesting change to the description to provide better clarity:

Provide an option to suppress error logging from mount_cb as some errors can be expected error and handled appropriately by datasources. For example: failure to mount ntfs volumes on VMs that do not have ntfs drivers.

File: tests/unittests/test_util.py [hunk 1]
```diff
@@ -1571,6 +1571,46 @@ def test_already_mounted_calls_callback_with_data(
...
+            "Failed to mount device: '/dev/sda1' with type: "
+            "'ntfs' using mount command:"
+        )
+        m_subp.side_effect = subp.ProcessExecutionError(
+            "", "unknown filesystem type 'ntfs'"
+        )
+        callback = mock.Mock(autospec=True)
+        with pytest.raises(Exception):
+            util.mount_cb(
+                "/dev/sda1",
```
Comment (cjp256) [inline tests/unittests/test_util.py:1586]: the tests may be failing if the system mounts are leaking and /dev/sda1 is already mounted?

**Comment by KsenijaS** [comment]
LGTM!

**Review by anhvoms — APPROVED** [review]

**Review by holmanb — APPROVED** [review]
@KsenijaS Thanks for proposing this change. This looks good to me.

> The only suggestion I have is to update test_mount_cb_does_not_log() to parametrize both cases (log_error=True and False) to ensure the log check is working as expected.
>
> Would be nice to add a test case for Azure as well to vet the expected behavior.

Do you plan on implementing the suggested changes by @cjp256?

**Comment by KsenijaS** [comment]
> @KsenijaS Thanks for proposing this change. This looks good to me.
> 
> > The only suggestion I have is to update test_mount_cb_does_not_log() to parametrize both cases (log_error=True and False) to ensure the log check is working as expected.
> > Would be nice to add a test case for Azure as well to vet the expected behavior.
> 
> Do you plan on implementing the suggested changes by @cjp256?

The first suggestion to parametrize both cases is done now.
For second one, I don't think there is a need for a test case for Azure, since the logs that we are trying to suppress are visible only from utils file.

**Pull request merged**